[2025-02-11T20:38:30.492+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-02-11T20:38:30.506+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: pyspark_test_work.execution manual__2025-02-11T20:38:28.818003+00:00 [queued]>
[2025-02-11T20:38:30.513+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: pyspark_test_work.execution manual__2025-02-11T20:38:28.818003+00:00 [queued]>
[2025-02-11T20:38:30.513+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 1
[2025-02-11T20:38:30.526+0000] {taskinstance.py:2889} INFO - Executing <Task(ClickHouseOperatorExtended): execution> on 2025-02-11 20:38:28.818003+00:00
[2025-02-11T20:38:30.534+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=152) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-02-11T20:38:30.535+0000] {standard_task_runner.py:72} INFO - Started process 154 to run task
[2025-02-11T20:38:30.535+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'pyspark_test_work', 'execution', 'manual__2025-02-11T20:38:28.818003+00:00', '--job-id', '325', '--raw', '--subdir', 'DAGS_FOLDER/bank_transactions.py', '--cfg-path', '/tmp/tmpt44wg7l4']
[2025-02-11T20:38:30.537+0000] {standard_task_runner.py:105} INFO - Job 325: Subtask execution
[2025-02-11T20:38:30.573+0000] {task_command.py:467} INFO - Running <TaskInstance: pyspark_test_work.execution manual__2025-02-11T20:38:28.818003+00:00 [running]> on host 334c275ebc16
[2025-02-11T20:38:30.658+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Smalch' AIRFLOW_CTX_DAG_ID='pyspark_test_work' AIRFLOW_CTX_TASK_ID='execution' AIRFLOW_CTX_EXECUTION_DATE='2025-02-11T20:38:28.818003+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-02-11T20:38:28.818003+00:00'
[2025-02-11T20:38:30.660+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-02-11T20:38:30.670+0000] {base.py:84} INFO - Retrieving connection 'clickhouse'
[2025-02-11T20:38:30.671+0000] {clickhouse.py:79} INFO - -- DROP TABLE IF EXISTS card_data.bank_transactions
[2025-02-11T20:38:30.768+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/lib/clickhouse_operator_extended.py", line 7, in execute
    return self._hook_execute()
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow_clickhouse_plugin/operators/clickhouse.py", line 69, in _hook_execute
    return hook.execute(
           ^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow_clickhouse_plugin/hooks/clickhouse.py", line 80, in execute
    last_result = conn.execute(
                  ^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/client.py", line 382, in execute
    rv = self.process_ordinary_query(
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/client.py", line 580, in process_ordinary_query
    return self.receive_result(with_column_types=with_column_types,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/client.py", line 212, in receive_result
    return result.get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/result.py", line 50, in get_result
    for packet in self.packet_generator:
                  ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/client.py", line 228, in packet_generator
    packet = self.receive_packet()
             ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/client.py", line 245, in receive_packet
    raise packet.exception
clickhouse_driver.errors.ServerException: Code: 62.
DB::Exception: Empty query. Stack trace:

0. DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000d74955b
1. DB::Exception::createDeprecated(String const&, int, bool) @ 0x000000000d7b708d
2. DB::parseQueryAndMovePosition(DB::IParser&, char const*&, char const*, String const&, bool, unsigned long, unsigned long, unsigned long) @ 0x00000000140dd332
3. DB::executeQueryImpl(char const*, char const*, std::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum, DB::ReadBuffer*) @ 0x00000000123d44a1
4. DB::executeQuery(String const&, std::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum) @ 0x00000000123d367d
5. DB::TCPHandler::runImpl() @ 0x00000000136b10fe
6. DB::TCPHandler::run() @ 0x00000000136cc378
7. Poco::Net::TCPServerConnection::start() @ 0x0000000016607027
8. Poco::Net::TCPServerDispatcher::run() @ 0x00000000166074b9
9. Poco::PooledThread::run() @ 0x00000000165d3ddc
10. Poco::ThreadImpl::runnableEntry(void*) @ 0x00000000165d237d
11. start_thread @ 0x0000000000007ea7
12. ? @ 0x00000000000fbacf

[2025-02-11T20:38:30.776+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=pyspark_test_work, task_id=execution, run_id=manual__2025-02-11T20:38:28.818003+00:00, execution_date=20250211T203828, start_date=20250211T203830, end_date=20250211T203830
[2025-02-11T20:38:30.788+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-02-11T20:38:30.789+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 325 for task execution (Code: 62.
DB::Exception: Empty query. Stack trace:

0. DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000d74955b
1. DB::Exception::createDeprecated(String const&, int, bool) @ 0x000000000d7b708d
2. DB::parseQueryAndMovePosition(DB::IParser&, char const*&, char const*, String const&, bool, unsigned long, unsigned long, unsigned long) @ 0x00000000140dd332
3. DB::executeQueryImpl(char const*, char const*, std::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum, DB::ReadBuffer*) @ 0x00000000123d44a1
4. DB::executeQuery(String const&, std::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum) @ 0x00000000123d367d
5. DB::TCPHandler::runImpl() @ 0x00000000136b10fe
6. DB::TCPHandler::run() @ 0x00000000136cc378
7. Poco::Net::TCPServerConnection::start() @ 0x0000000016607027
8. Poco::Net::TCPServerDispatcher::run() @ 0x00000000166074b9
9. Poco::PooledThread::run() @ 0x00000000165d3ddc
10. Poco::ThreadImpl::runnableEntry(void*) @ 0x00000000165d237d
11. start_thread @ 0x0000000000007ea7
12. ? @ 0x00000000000fbacf
; 154)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/lib/clickhouse_operator_extended.py", line 7, in execute
    return self._hook_execute()
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow_clickhouse_plugin/operators/clickhouse.py", line 69, in _hook_execute
    return hook.execute(
           ^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow_clickhouse_plugin/hooks/clickhouse.py", line 80, in execute
    last_result = conn.execute(
                  ^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/client.py", line 382, in execute
    rv = self.process_ordinary_query(
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/client.py", line 580, in process_ordinary_query
    return self.receive_result(with_column_types=with_column_types,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/client.py", line 212, in receive_result
    return result.get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/result.py", line 50, in get_result
    for packet in self.packet_generator:
                  ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/client.py", line 228, in packet_generator
    packet = self.receive_packet()
             ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/client.py", line 245, in receive_packet
    raise packet.exception
clickhouse_driver.errors.ServerException: Code: 62.
DB::Exception: Empty query. Stack trace:

0. DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000d74955b
1. DB::Exception::createDeprecated(String const&, int, bool) @ 0x000000000d7b708d
2. DB::parseQueryAndMovePosition(DB::IParser&, char const*&, char const*, String const&, bool, unsigned long, unsigned long, unsigned long) @ 0x00000000140dd332
3. DB::executeQueryImpl(char const*, char const*, std::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum, DB::ReadBuffer*) @ 0x00000000123d44a1
4. DB::executeQuery(String const&, std::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum) @ 0x00000000123d367d
5. DB::TCPHandler::runImpl() @ 0x00000000136b10fe
6. DB::TCPHandler::run() @ 0x00000000136cc378
7. Poco::Net::TCPServerConnection::start() @ 0x0000000016607027
8. Poco::Net::TCPServerDispatcher::run() @ 0x00000000166074b9
9. Poco::PooledThread::run() @ 0x00000000165d3ddc
10. Poco::ThreadImpl::runnableEntry(void*) @ 0x00000000165d237d
11. start_thread @ 0x0000000000007ea7
12. ? @ 0x00000000000fbacf

[2025-02-11T20:38:30.829+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-02-11T20:38:30.848+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-02-11T20:38:30.851+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
